src/tools/common.ts

```diff
@@ -95,8 +98,9 @@ export const wait: Tool = {
   },
   handle: async (context, params) => {
     const validatedParams = waitSchema.parse(params);
+    const maxTimeMs = (validatedParams.maxTime || 10) * 1000;
     const page = await context.ensurePage();
-    await page.waitForTimeout(Math.min(10000, validatedParams.time * 1000));
+    await page.waitForTimeout(Math.min(maxTimeMs, validatedParams.time * 1000));
```

Comment:
waitForTimeout waits for given time, there is no point in taking time and maxTime from the user. The maxTime cap here is to ensure no stalls.